Snippets/MsmqTransport/MsmqTransport_All/QueueCreation/QueueCreationUtils.cs

```diff
@@ -12,21 +12,16 @@ public static class QueueCreationUtils
         public static void CreateQueue(string queueName, string account)
         {
             var path = $@"{Environment.MachineName}\private$\{queueName}";
-            if (MessageQueue.Exists(path))
+            if (!MessageQueue.Exists(path))
             {
-                using (var messageQueue = new MessageQueue(path))
+                using (var messageQueue = MessageQueue.Create(path, true))
                 {
-                    SetPermissionsForQueue(messageQueue, account);
-                    return;
+                    SetDefaultPermissionsForQueue(messageQueue, account);
                 }
             }
-            using (var messageQueue = MessageQueue.Create(path, true))
-            {
-                SetPermissionsForQueue(messageQueue, account);
-            }
         }
 
-        static void SetPermissionsForQueue(MessageQueue queue, string account)
+        static void SetDefaultPermissionsForQueue(MessageQueue queue, string account)
```

Comment:
No need to set anonymous + everyone since MSMQ sets that by default ever since Server 2003

Not sure but I think admin is given fullcontrol by default as well?